packages/dds/sequence/src/sharedStringWithSequentialId.ts

```diff
@@ -0,0 +1,84 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { Lazy } from "@fluidframework/common-utils";
+import { IChannelAttributes, IFluidDataStoreRuntime } from "@fluidframework/datastore-definitions";
+import { Marker, MergeTreeDeltaType, MergeTreeMaintenanceType,
+     reservedMarkerIdKey, SortedSegmentSet } from "@fluidframework/merge-tree";
+import { createIdBetween, digitLast, digitZero } from "./generateSequentialId";
+import { SequenceDeltaEvent, SequenceMaintenanceEvent } from "./sequenceDeltaEvent";
+import { SharedString } from "./sharedString";
+
+export function sharedStringWithSequentialIdMixin(Base: typeof SharedString = SharedString): typeof SharedString {
```

Comment:
I'd look at mixin as a tool to get what you want.
But in terms of packaging (i.e. how it's exposed to outside world), it needs to be its own DDS with its own factory.
I.e. we need to override getFactory() and return new factory (that is almost a copy of SharedStringFactory) that points back to this class.
The new factory will be "fixed" i.e. it will use sharedStringWithSequentialIdMixin(SharedString) (not some other class).
Mixin model just allows us in the future (if we need to) create more of these things easily, potentially extending classes like SequenceWithSequentialId.

Basically Scriptor would need to use that new DDS and every we would know how to load it (that specific DDS) given it has different (new) factory type name. But someone else may not use this class and continue to use SharedString itself, and when we load such DDS, we will instantiate Sequence, not SequenceWithSequentialId for such users.

Let me know if it does not make sense :)

Reply:
Added the factory. Do you guys think it makes sense to move this to another package? 
@anthony-murphy as well. At first i thought there was a sharedStringWithInterception already as well inside sequence. But given that I am creating a very specialized factory for sequentialId I am open to this now

Reply:
Spoke with vlad offline and I moved this to a new package called sequence-sequential-id inside dds. I am open to feedback on naming thought